TaskEngine/Tasks/CleanUpElasticIndexTask.cs

```diff
@@ -40,47 +39,49 @@ protected async override Task OnConsume(string example, TaskParameters taskParam
             registerTask(cleanup, "CleanUpElasticIndexTask"); // may throw AlreadyInProgress exception
             _logger.LogInformation("CleanUpElasticIndexTask Starting");
 
-            //using (var _context = CTDbContext.CreateDbContext())
-            //{
-            //    CaptionQueries captionQueries = new CaptionQueries(_context);
-
-            //    var all_transcriptions = await _context.Transcriptions.Where(t => t.Language == Languages.ENGLISH_AMERICAN).ToListAsync();
-            //    foreach (var transcription in all_transcriptions)
-            //    {
-            //        var all_captions = transcription.Captions;
+            var result = await _client.Indices.GetAsync(new GetIndexRequest(Indices.All));
+            var indices = result.Indices;
+            foreach (var index in indices)
+            {
+                string index_name = index.Key.ToString();
 
-            //        // each index has the unique name "index_string_unique", the current in use one has the alias "index_string_alias"
-            //        var index_string_base = transcription.Id + "_" + Languages.ENGLISH_AMERICAN.ToLower();
-            //        var index_string_unique = index_string_base + "_" + $"{DateTime.Now:yyyyMMddHHmmss}";
-            //        var index_string_alias = index_string_base + "_" + "primary";
+                // elastic search internal management index starts with '.'
+                if (index_name[0] == '.')
+                {
+                    continue;
+                }
 
-            //        var asyncBulkIndexResponse = await _client.BulkAsync(b => b
-            //            .Index(index_string_unique)
-            //            .IndexMany(all_captions)
-            //        );
+                _logger.LogInformation(index.Key.ToString());
+                string[] parts = index_name.Split("_");
+                var index_string_id = parts[0];
+                var index_string_lang = parts[1];
+                var index_string_time = parts[2];
 
-            //        var alias_exist = await _client.Indices.ExistsAsync(index_string_alias);
-            //        if (alias_exist.Exists)
-            //        {
-            //            var oldIndices = await _client.GetIndicesPointingToAliasAsync(index_string_alias);
-            //            var oldIndexName = oldIndices.First().ToString();
+                // Indices that were created in the last 2 days would not be removed.
+                DateTime createdAt = DateTime.ParseExact(index_string_time, "yyyyMMddHHmmss", null);
+                _logger.LogInformation("Created at: " + createdAt.ToString());
+                DateTime range = DateTime.Now.AddDays(-2);
+                if (DateTime.Compare(createdAt, range) >= 0)
+                {
+                    _logger.LogInformation("Skipped: Index is created in the last two days");
```

Comment:
(This might be rather verbose on the production server. do you really need this?)

Reply:
Oh, I thought we are gonna run this on dev for a few days. It's helpful to figure out why an index is not removed. Shall I delete those two?